### PR TITLE
Weak pawns improved version (should compile) 

### PR DESCRIPTION
**Pull request by abaiesu**
Fix compilation errors and tested the whole weakPawns.cpp file. PieceMovement.cpp should compile but now I have a problem with the test file weak_paws_tests.cpp because the ".exe" files doesn't exist anymore. If anyone has any clue on how to fix that and run  weak_paws_tests.cpp let me know.

### CODE REVIEW

**Comment by abaiesu** [comment]
All files compile, there seems to be no problems in both weakPawns.cpp and PieceMovement.cpp. Both files have functions that pass basic tests.

**Comment by plumsirawit** [comment]
This is mergeable but I'd like to have a review from @souhhcong to confirm the changes. Thanks @abaiesu.

File: lib/weak_pawns/PieceMovements.cpp [hunk 1]
```diff
@@ -14,91 +14,29 @@ using namespace WeakPawns;
...
-    uint64_t dangerBoard = 0;
+    uint64_t dangerBoard = 0ll;
     bool opp_color = (color+1)%2; //get the opponent's color
 
     for(int t = 0; t<=63; t++){ //we check each tile
         int piece = board.getPieceFromCoords(t);
-        if((piece%2 != -1) && (piece%2 == opp_color)){ //we found an opponent piece
-            dangerBoard = dangerBoard | (board.getLegalMoves(piece, t) & board.getColorBitBoard(opp_color));
+        if((piece != -1) && (piece%2 == opp_color)){ //we found an opponent piece
+            dangerBoard += board.getLegalMoves(piece, t);
```
Comment (plumsirawit) [inline lib/weak_pawns/PieceMovements.cpp:23]: Why use addition on bitboard?

**Review by plumsirawit — APPROVED** [review]
Looks good! Thanks!

**Review by souhhcong — APPROVED** [review]
merge and inshallah

**Pull request merged**